controllers/articles.js

```diff
@@ -190,23 +204,49 @@ const articles = {
         offset
       });
 
-      response.forEach((element) => {
-        element.ratings = getAverageRating(element);
-        element.author = authorInfo;
-      });
+      const formatedArticles = await Promise.all(
+        allPublishedArticles.map(async (publishedArticle) => {
+          // Get average ratings for each article
+          const ArticleRatings = await ratings.findAll({
+            where: { post: publishedArticle.id },
+            attributes: ['rating']
+          });
+
+          const articleLikes = await likes.findAndCountAll({
+            where: { articleId: publishedArticle.id, status: 'liked' }
+          });
+          const articleRatings = getAverageRating(ArticleRatings);
+          publishedArticle.likes = articleLikes.count;
+          publishedArticle.ratings = articleRatings;
+
+          return {
+            title: publishedArticle.title,
+            body: publishedArticle.body,
+            description: publishedArticle.description,
+            slug: publishedArticle.slug,
+            tagList: publishedArticle.tagList,
+            ratings: publishedArticle.ratings,
```

Comment:
For avoiding repetition of `publishedArticle.x` It's better if you destructure publishedArticle object then make on object literal (ES6) on the return response.